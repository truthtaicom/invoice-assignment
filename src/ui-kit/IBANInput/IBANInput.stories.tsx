import { storiesOf } from '@storybook/react';
import React from 'react';
import IBANInput from "./IBANInput"

storiesOf("UI-Kit", module)
  .add("IBANInput", () => (
    <IBANInput />
  ))
