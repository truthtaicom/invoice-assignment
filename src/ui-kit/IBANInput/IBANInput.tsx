import React from 'react';
import InputMask from 'react-input-mask';
import { Input } from '../'

const IBANInput = (props) => {
 return (
  <InputMask {...props} mask="DE-99-9999-9999-99999-99">
    {(inputProps) => (
      <Input {...inputProps} placeholder="DE-12-1234-1234-12345-12" />
    )}
  </InputMask>
 );
}

export default IBANInput
