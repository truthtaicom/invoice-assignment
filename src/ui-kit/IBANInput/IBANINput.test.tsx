import React from 'react'
import { render, fireEvent } from 'react-testing-library'
import IBANInput from './IBANInput'

describe('<IBANInput /> spec', () => {
  it('renders the component', () => {
    const component = render(<IBANInput value="DE-12-1234-1234-12345-12" />)
    expect(component).toMatchSnapshot()
  })
});