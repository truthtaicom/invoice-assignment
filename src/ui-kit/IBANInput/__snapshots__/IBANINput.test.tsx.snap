// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IBANInput /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-htpNat ctuoOm"
      >
        <div
          class="sc-bxivhb cWjgJW"
        >
          <span
            class="sc-gzVnrw gtppSg"
          />
          <span
            class="sc-gzVnrw gtppSg"
          >
            DE-12-1234-1234-12345-12
             
          </span>
          <span
            class="sc-gzVnrw gtppSg"
          >
            +
          </span>
        </div>
        <div
          class="sc-bxivhb hWYmZL"
        >
          <input
            class="sc-bZQynM htDgzj"
            placeholder="DE-12-1234-1234-12345-12"
            value="DE-12-1234-1234-12345-12"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-htpNat ctuoOm"
    >
      <div
        class="sc-bxivhb cWjgJW"
      >
        <span
          class="sc-gzVnrw gtppSg"
        />
        <span
          class="sc-gzVnrw gtppSg"
        >
          DE-12-1234-1234-12345-12
           
        </span>
        <span
          class="sc-gzVnrw gtppSg"
        >
          +
        </span>
      </div>
      <div
        class="sc-bxivhb hWYmZL"
      >
        <input
          class="sc-bZQynM htDgzj"
          placeholder="DE-12-1234-1234-12345-12"
          value="DE-12-1234-1234-12345-12"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
